### PR TITLE
Support for python virtualenv in compile/test scripts

### PR DESCRIPTION
**Pull request by alexbaden**
I switched from conda to virtualenv on the PVC lab machine due to include errors w/ system libraries for level0. These changes allow me to continue using the repo build and test scripts w/out breaking existing workflows.

### CODE REVIEW

File: scripts/compile-triton.sh [hunk 1]
```diff
@@ -119,6 +135,14 @@ fi
 function build_triton {
   echo "**** Configuring $TRITON_PROJ ****"
   cd $TRITON_PROJ
+
+  if [ "$VENV" = true ]; then
+    echo "**** Creating Python virtualenv ****"
+    python3 -m venv .venv --prompt triton
+    source .venv/bin/activate
+    pip install ninja cmake wheel
```
Comment (ienkovich) [inline scripts/compile-triton.sh:143]: If you want to use cmake from the virtual env then you need to fix the CMAKE definition which is now unconditionally `CMAKE=/usr/bin/cmake`

Comment (alexbaden) [reply scripts/compile-triton.sh:143]: tbh I copied this directly from triton docs: https://github.com/openai/triton?tab=readme-ov-file#install-from-source 

b/c build_triton happens after build_llvm we wouldn't be able to use this cmake in the llvm stage. I don't know what cmake we use for triton, but it doesn't seem to matter.

**Review by pbchekin — APPROVED** [review]
Not sure why you need this in the scripts since you always can create/initialize venv before executing them, but lgtm.

**Comment by alexbaden** [comment]
> Not sure why you need this in the scripts since you always can create/initialize venv before executing them, but lgtm.

Well, we need wheel in the virtual environment but don't need it if you're using some existing python env. Theoretically anyway. Note that in the build script you install wheel during the build triton step, but that should probably be in the compile-triton script. 

There's another argument which is to always use the venv so we have a standardized build env, but I don't know how that would work for people using conda. On fox127 we have to use venv if we don't use conda - at least that has been @whitneywhtsang and my experience.

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**